internal/service/config/nginx_config_service.go

```diff
@@ -147,16 +169,98 @@ func (n *Nginx) Validate() error {
 
 func (n *Nginx) Apply() error {
 	slog.Debug("Applying NGINX config")
+	var errorsFound []string
```

Comment:
Is there a reason why we use `[]string` for this instead of `[]error`? We could just use `errors.Join()` to collect errors, so something like:
```go
var errorsFound error

// monitor logs and reload

for i := 0; i < numberOfExpectedMessages; i++ {
	err := <-logErrorChannel
	slog.Debug("Message received in logErrorChannel", "error", err)
	if err != "" {
		errorsFound = errors.Join(errorsFound, err)
	}
}

slog.Info("Finished monitoring post reload")

if errorsFound != nil {
  	return errorsFound // This would have all errors, instead of just the first one.
}
```

Reply:
that looks like a cleaner way to do it alright. Let me make that update